src/analog_temp.c

```diff
@@ -0,0 +1,113 @@
+/*
+
+	FILENAME: 			analog_temp.c
+	DEPENDENCIES:		spi, pgmspace, delay, stdint
+
+	DESCRIPTION:		Functions for measuring thermistor temperature
+	AUTHORS:			Dylan Vogel, James Sheridan
+	DATE MODIFIED:	    2017-12-29
+
+	NOTE:               TODO: Implement more efficient channel reading algorithm,
+                        instead of having to waste SPI frames from waiting for
+                        next message
+
+	REVISION HISTORY:
+
+        2017-12-29      DV: Implemented lookup table with flash memory
+        2017-12-28      DV: Created file, implemented lookup table, resistance
+                        calculation.
+
+*/
+
+#include "analog_temp.h"
+
+void init_thermistor(void){
+    // Initialize COM_RST pin on 32M1 as ADC CS
+
+    init_cs(COM_RST, &COM_RST_DDR);
+    set_cs_high(COM_RST, &COM_RST_PORT);
+}
+
+uint16_t read_thermistor_adc(int channel){
+    // Returns the raw ADC value for the specified channel
+    // Assumes channel is one of 0-6
+    uint16_t data;
+    uint8_t ADC_cmd, channel_bits;
+
+    if (channel % 2 == 0){
+        channel_bits = (uint8_t)(channel / 2);
+    } else {
+        channel_bits = 0b100  | (uint8_t)(channel / 2);
```

Comment:
Not usually a good idea to do integer division like this. If `channel` is odd, `channel / 2` is always `1 = 0b001`. Thus `channel_bits` is always `0b101` in this case. Is this the intended behaviour?

Reply:
No, the intended behavior is as follows (pg 10):
<img width="512" alt="image" src="https://user-images.githubusercontent.com/23109037/34450510-e9f2016e-ecd8-11e7-8225-48a5cf607bc6.png">

Reply:
Shouldn't `channel/2` return `0b000` for 1, `0b001` for 3, `0b010` for 5 ...?

Reply:
Oh ofcourse; thanks, I'm being an idiot.